channel/src/main/java/com/synopsys/integration/alert/channel/email/actions/EmailTestActionHelper.java

```diff
@@ -24,72 +24,68 @@
-            Set<ProviderProject> providerProjects = retrieveProviderProjects(fieldUtility, filterByProject, providerConfigId);
-            if (null != providerProjects) {
-                Set<String> providerEmailAddresses = addEmailAddresses(providerConfigId, providerProjects, fieldUtility);
+            Set<ProviderProject> providerProjects = retrieveProviderProjects(distributionJobModel, providerConfigId);
+            if (null != providerProjects && !providerProjects.isEmpty()) {
+                Set<String> providerEmailAddresses = addEmailAddresses(providerConfigId, providerProjects, distributionJobModel, emailJobDetails);
                 emailAddresses.addAll(providerEmailAddresses);
             }
         }
-
-        ConfigurationFieldModel configurationFieldModel = ConfigurationFieldModel.create(EmailDescriptor.KEY_EMAIL_ADDRESSES);
-        configurationFieldModel.setFieldValues(emailAddresses);
-
-        Map<String, ConfigurationFieldModel> fields = fieldUtility.getFields();
-        fields.put(EmailDescriptor.KEY_EMAIL_ADDRESSES, configurationFieldModel);
-
-        return new FieldUtility(fields);
+        return emailAddresses;
     }
 
     private boolean doesProjectMatchConfiguration(String currentProjectName, String projectNamePattern, Set<String> configuredProjectNames) {
         return currentProjectName.matches(projectNamePattern) || configuredProjectNames.contains(currentProjectName);
     }
 
-    private Set<ProviderProject> retrieveProviderProjects(FieldUtility fieldUtility, boolean filterByProject, Long providerConfigId) {
+    private Set<ProviderProject> retrieveProviderProjects(DistributionJobModel distributionJobModel, Long providerConfigId) {
         List<ProviderProject> providerProjects = providerDataAccessor.getProjectsByProviderConfigId(providerConfigId);
-        if (filterByProject) {
-            Optional<ConfigurationFieldModel> projectField = fieldUtility.getField(ProviderDistributionUIConfig.KEY_CONFIGURED_PROJECT);
-            Set<String> configuredProjects = new HashSet<>(projectField.map(ConfigurationFieldModel::getFieldValues).orElse(Set.of()));
-            String projectNamePattern = fieldUtility.getStringOrEmpty(ProviderDistributionUIConfig.KEY_PROJECT_NAME_PATTERN);
+        if (distributionJobModel.isFilterByProject()) {
+            Set<String> configuredProjects = distributionJobModel.getProjectFilterDetails()
+                                                 .stream()
+                                                 .map(BlackDuckProjectDetailsModel::getName)
+                                                 .collect(Collectors.toSet());
+            String projectNamePattern = distributionJobModel.getProjectNamePattern().orElse("");
             return providerProjects
                        .stream()
                        .filter(databaseEntity -> doesProjectMatchConfiguration(databaseEntity.getName(), projectNamePattern, configuredProjects))
```

Comment:
I don't think the ProviderProject's are database entities anymore, this should be updated to reflect that.